app/assets/javascripts/user_analytics.js

```diff
@@ -0,0 +1,185 @@
+//
+// stats
+//
+function statistics() {
+  return JSON.parse(document
+    .getElementById('statistics')
+    .attributes
+    .getNamedItem('data-statistics')
+    .value);
+}
+
+function dailyStatistics() {
+  return Object.entries(statistics()
+    .daily
+    .grouped_by_date);
+}
+
+//
+// dates
+//
+function formatDate(date) {
+  return date.toJSON().slice(0, 10);
+}
+
+function today() {
+  return new Date();
+}
+
+function yesterday() {
+  var todayDate = today();
+  return new Date(todayDate.setDate(todayDate.getDate() - 1));
+}
+
+function formattedTomorrowDate() {
+  return statistics().metadata.formatted_next_date;
+}
+
+function formattedTodayString() {
+  return statistics().metadata.formatted_today_string;
+}
+
+function dailyStatisticsDates() {
+  var dates = [];
+
+  for (let [_, item] of dailyStatistics()) {
+    dates.push(item)
+  }
+
+  return Object.keys(Object.assign(...dates));
+}
+
+function latestDateInDailyStatistics() {
+  var dates = [];
+
+  for (let date of dailyStatisticsDates()) {
+    dates.push(Date.parse(date));
+  }
+
+  return new Date(Math.max.apply(null, dates));
+}
+
+//
+// elements
+//
+function syncNudgeCardElement() {
+  return document.querySelector('#daily-stats-card > .count-empty');
+}
+
+function allDaysInCarouselElements() {
+  return document.getElementsByClassName("day");
+}
+
+function nextSlideButtonElement() {
+  return document.getElementsByClassName("button-next")[0];
+}
+
+function prevSlideButtonElement() {
+  return document.getElementsByClassName("button-prev")[0];
+}
+
+function syncNudgeCardPosition() {
+  for (let [position, element] of Object.entries(allDaysInCarouselElements())) {
+    if (element === syncNudgeCardElement()) {
```

Comment:
Maybe we should get the `syncNudgeCardElement` just once at the beginning of this method?

Reply:
Addressed JS things here: https://github.com/simpledotorg/simple-server/pull/843/commits/ad4d217691c22b825a7a76e43af3099e9be2fa03. This includes moving to ES6 best practices (no usage of `var`s anywhere).